AzureIoTHub.agent.lib.nut

```diff
@@ -629,513 +632,819 @@ class AzureIoTHub {
+                "twinUpdParallelReqs" : AZURE_IOT_CLIENT_DEFAULT_TWIN_UPD_PARAL_REQS,
+                "sendMsgParallelReqs" : AZURE_IOT_CLIENT_DEFAULT_MSG_SEND_PARAL_REQS
+            };
 
-        // done cb params - err
-        function sendEvent(message, done = null) {
-            if ( !_isOpen(_sessions.event) ) {
-                if (done) done("Cannot send while disconnected.");
-            } else {
-                if ( !_isOpen(_senders.event) ) {
-                    _log("creating event sender");
-                    // add message to queue
-                    _msgQueue.push({"msg" : message, "cb" : done});
-                    // create a new sender & send message
-                    _openEventSender();
-                } else {
-                    _log("event sender open")
-                    _sendEvent(message, done);
-                }
-            }
-        }
+            _onConnectCb      = onConnect;
+            _onDisconnectCb   = onDisconnect;
 
-        // done is called everytime a message is available, params - error, data
-        function receive(done) {
-            _handlers.onEvent <- done;
+            _connStrParsed = AzureIoTHub.ConnectionString.Parse(deviceConnStr);
+            _mqttclient = mqtt.createclient();
+            _mqttclient.onconnect(_onConnect.bindenv(this));
+            _mqttclient.onconnectionlost(_onDisconnect.bindenv(this));
+            _mqttclient.onmessage(_onMessage.bindenv(this));
 
-            if (done == null) {
-                _receivers.event.close();
-                // clear out handlers
-                return this;
+            foreach (optName, optVal in options) {
+                _options[optName] <- optVal;
             }
 
-            if( !_isOpen(_sessions.event) ) {
-                _handlers.onEvent("Cannot receive while disconnected.", null);
-            } else {
-                // TODO: test/handle if receiver already open
-                if ( _isOpen(_receivers.event) ) _receivers.event.close();
-                _openEventReceiver();
-            }
-            return this;
+            _fillTopics(_connStrParsed.DeviceId);
         }
 
-        // PRIVATE FUNCTIONS
-        // ------------------------------------------------------------------------------------
-        // ------------------------------------------------------------------------------------
+        // Opens a connection to Azure IoT Hub.
+        //
+        // Returns:                         Nothing.
+        function connect() {
+            if (_state == AZURE_IOT_CLIENT_STATE_DISCONNECTED) {
+                _log("Connecting...");
 
-        // Open Auth/Event Session/Sender/Receivers
-        // ------------------------------------------------------------------------------------
+                local devPath = "/" + _connStrParsed.DeviceId;
+                local username = _connStrParsed.HostName + devPath + format("/api-version=%s", AZURE_API_VERSION);
```

Comment:
what's the point of using `format` if you do concatenation around it? why not to use it for the entire string or just don't use at all?

Reply:
applicable for the next line as well

Reply:
Done